doc/Language/classtut.pod6

```diff
@@ -484,12 +484,96 @@ X<|FQN>
 
 Just like attributes, methods can also be private. Private methods are declared
 with a prefixed exclamation mark. They are called with C<self!>
-followed by the method's name. To call a private method of another class the
-calling class has to be trusted by the called class. A trust relationship is
-declared with C<trusts> and the class to be trusted must already be declared.
-Calling a private method of another class requires an instance of that class
-and the fully qualified name (FQN) of the method. Trust also allows access to
-private attributes.
+followed by the method's name. In the following implementation of a C<Region>
+class, methods C<population-info> and C<densely-populated> are private methods,
+while the rest are public ones:
+
+=begin code
+class Region {
+    has $.name;
+    has $.population;
+    has $.area-size;
+    has $.continent;
+    has $.high-population-density;
+
+    method info {
+        say "Name: $!name";
+        say self!population-info;
+    }
+
+    method more-densely-populated(Region:D $r) {
+        my $result = self.population-density > $r.population-density ?? 'more' !! 'less';
+        say "!name is $result densely populated than $r..name.";
+    }
+
+    method population-density {
+        ($!population / $!area-size).round(0.01)
+    }
+
+    method can-be-crowded {
+        self!densely-populated
+        ?? "$!name can be crowdy."
+        !! "There is enough space in $!name."
+    }
+
+    method !population-info(--> Str:D) {
+        "$!population people live here."
+    }
+
+    method !densely-populated(--> Bool:D) {
+        self.population-density > $!high-population-density
+    }
+}
+
+my constant city-hpd = 3_000;
+
+my $wroclaw = Region.new(
+    :name<Wrocław>,
+    :population(638_000),
+    :area-size(293),
+    :continent<Europe>,
+    :high-population-density(city-hpd)
+);
+=end code
+
+Regular methods can be introspected with C<.^methods> and resolved with
+C<.^find-method>.
+
```

Comment:
You should probably clarify what it actually returns, not show the result of say over them. Also, link to them.